### PR TITLE
[e2e tests]: Address flake in test_id:4643

### PR DESCRIPTION
**Pull request by akalenyu**
Signed-off-by: Alex Kalenyuk <akalenyu@redhat.com>

**What this PR does / why we need it**:
Not sure exactly what goes wrong here, at first, I thought it was a low timeout (nanoseconds) but apparently, gomega will convert integers to seconds.
Possibly the double return value paired with a single matcher? https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/1695
Regardless, I think this reads a little better.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Review by xpivarc — COMMENTED** [review]
/approve

File: tests/storage/datavolume.go [hunk 1]
```diff
@@ -404,16 +404,17 @@ var _ = SIGDescribe("DataVolume Integration", func() {
...
 
 			It("[test_id:4643]should NOT be rejected when VM template lists a DataVolume, but VM lists PVC VolumeSource", func() {
 				_, err = virtClient.CdiClient().CdiV1beta1().DataVolumes(util.NamespaceTestDefault).Create(context.Background(), dv, metav1.CreateOptions{})
 				Expect(err).ToNot(HaveOccurred())
-				defer libstorage.DeleteDataVolume(&dv)
 
-				Eventually(func() (*k8sv1.PersistentVolumeClaim, error) {
-					return virtClient.CoreV1().PersistentVolumeClaims(util.NamespaceTestDefault).Get(context.Background(), dv.Name, metav1.GetOptions{})
-				}, 30).Should(Not(BeNil()))
+				Eventually(func() error {
```
[the remark this answers is not in the document]
Comment (akalenyu) [reply tests/storage/datavolume.go:414]: We can but I would keep this as is, DV succeeded is not exactly mandatory here to proceed

**Comment by akalenyu** [comment]
/hold
until we're sure its fixed

**Comment by akalenyu** [comment]
@xpivarc updated the PR description, passing an integer to eventually should get converted to seconds apparently.
So, I'm not sure why it seems that this PR gets rid of the flake.

**Comment by xpivarc** [comment]
> @xpivarc updated the PR description, passing an integer to eventually should get converted to seconds apparently. So, I'm not sure why it seems that this PR gets rid of the flake.

I missed the fact there is already a timeout specified.

**Comment by akalenyu** [comment]
> > @xpivarc updated the PR description, passing an integer to eventually should get converted to seconds apparently. So, I'm not sure why it seems that this PR gets rid of the flake.
> 
> I missed the fact there is already a timeout specified.

OK, I think it might indeed be the double return value paired with a single matcher.
People have seen this before, for example:
https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/1695

**Comment by akalenyu** [comment]
/hold cancel
/test pull-kubevirt-e2e-k8s-1.22-sig-storage
unrelated failure

**Review by ShellyKa13 — COMMENTED** [review]
/lgtm

**Pull request merged**